### PR TITLE
[APK] Add API Key Authentication to agent

### PR DESCRIPTION
**Pull request by sgayangi**
## Purpose
Currently, the CP to DP flow for APK does not support API Key Authentication.
This PR introduces API Key Authentication as a security scheme so that API Key authentication functions as expected.

Fixes https://github.com/wso2/apk/issues/2311

### CODE REVIEW

**Review by CrowleyRajapakse — APPROVED** [review]

File: apim-apk-agent/pkg/transformer/transformer.go [hunk 1]
```diff
@@ -384,14 +384,14 @@ func getReqAndResInterceptors(reqPolicyCount, resPolicyCount int, reqPolicies []
 
 // mapAuthConfigs will take the security schemes as the parameter and will return the mapped auth configs to be
 // added into the apk-conf
-func mapAuthConfigs(apiUUID string, authHeader string, secSchemes []string, certAvailable bool, certList CertDescriptor, apiUniqueID string) []AuthConfiguration {
+func mapAuthConfigs(apiUUID string, authHeader string, configuredAPIKeyHeader string, secSchemes []string, certAvailable bool, certList CertDescriptor, apiUniqueID string) []AuthConfiguration {
```
Comment (Tharanidk) [inline apim-apk-agent/pkg/transformer/transformer.go:387]: It seems that this line exceeds the character limit

File: apim-apk-agent/pkg/transformer/transformer.go [hunk 2]
```diff
@@ -174,7 +174,7 @@ func GenerateAPKConf(APIJson string, certArtifact CertificateArtifact, organizat
 		certAvailable = true
 	}
 
-	authConfigList := mapAuthConfigs(apiYamlData.ID, apiYamlData.AuthorizationHeader, apiYamlData.SecuritySchemes, certAvailable, certList, apiUniqueID)
+	authConfigList := mapAuthConfigs(apiYamlData.ID, apiYamlData.AuthorizationHeader, apiYamlData.APIKeyHeader, apiYamlData.SecuritySchemes, certAvailable, certList, apiUniqueID)
```
Comment (Tharanidk) [inline apim-apk-agent/pkg/transformer/transformer.go:177]: this line exceeds the character limit

File: apim-apk-agent/pkg/transformer/transformer.go (hunk 1 above)
Comment (sgayangi) [reply apim-apk-agent/pkg/transformer/transformer.go:387]: Fixed in PR https://github.com/wso2/product-apim-tooling/pull/1168

File: apim-apk-agent/pkg/transformer/transformer.go (hunk 2 above)
Comment (sgayangi) [reply apim-apk-agent/pkg/transformer/transformer.go:177]: Fixed in PR https://github.com/wso2/product-apim-tooling/pull/1168

**Pull request merged**